### ISSUE TITLE
Use Cmd '+' as keybinding for "Zoom In" on OS X

### ISSUE THREAD

**Issue by weinand** [Microsoft/vscode#1240]
Currently it is Cmd+'=' which is not intuitive.

**Comment by bpasero** [comment]
Hm it should be...

**Comment by bpasero** [comment]
So Cmd++ works for me, which is cool. However what is weird is that we set the keybinding like this:

`{ primary: KeyMod.CtrlCmd | KeyCode.US_EQUAL }`

I could not find KeyCode.US_PLUS. 

@alexandrudima can you comment on this maybe?

**Comment by weinand** [comment]
@bpasero does Cmd++ works for you on a non-US keyboard?

**Comment by bpasero** [comment]
@weinand yes I am on german keyboard layout (ibm) and interestingly when I type "Cmd++" into the keyboard widget, I see it prints Cmd+= for me. What does it print for you?

**Comment by weinand** [comment]
@bpasero I'm on a Swiss German keyboard and I see "shift+cmd+=" and I know where the '=' comes from. On a US keyboard the '+' and the '=' are together on a key and since we only have KeyCode constants for the unshifted key, it is KeyCode.US_EQUAL.

@bpasero restating my question from above: did Cmd++ work for you **without** changing the keybinding?

**Comment by bpasero** [comment]
@weinand yes always worked for me without changing anything. and I see that we did not change this keybinding in months.

**Comment by weinand** [comment]
@bpasero yes, it worked for you because on a German keyboard the '+' doesn't require a 'Shift' modifier. For a Swiss-German keyboard it never worked.

**Comment by bpasero** [comment]
@weinand that is odd, I leave it to @alexandrudima if we can provide the keybinding that works for Swiss and other keyboard layouts, but we might be limited here because we cannot express per-language keybindings.

**Comment by weinand** [comment]
@bpasero after discussions with @alexandrudima I now better understand the issues around this problem and yes, we came to the same conclusion that we can only solve the issue with keyboard layout specific keybindings.

**Comment by patrys** [comment]
I think it's maybe time to revive this issue. A number of other issues have been merged into this one including a lot of issues about languages that depend on AltGr. Now AltGr is a technique that gives one or both <kbd>Alt</kbd> a special function of switching to a secondary keyboard layout group.

Polish for example depends on AltGr to to switch between naked characters and accented ones. Pressing <kbd>Z</kbd> gives you `Z` while pressing <kbd>Alt</kbd>+<kbd>Z</kbd> gives you `Ż`. Polish diacritics are not otherwise available as separate keys (long story short: there used to be a Polish typist keyboard layout and typewriters used it but we had to import computers with US keyboards because communism).

Now <kbd>Alt</kbd>+<kbd>Z</kbd> from the example above is currently mapped to "toggle word wrap". This means we can't type in Polish without remapping parts of the keyboard.

**Comment by alexandrudima** [comment]
@patrys On mac we ship with 516 default keybindings. There are hundreds of possible keyboard layouts only on the mac and it is possible for people to customize them and create their own keyboard layouts.

It is difficult to find key combinations which are both easy enough to be remembered, easy enough to be ergonomically feasible and which work across a wide number of the keyboard layouts well.

The correct fix for your case is for VS Code to ship with <kbd>Alt</kbd>+<kbd>Z</kbd> as the keybinding for toggle word wrap for every keyboard layout except the Polish one, under which <kbd>Alt</kbd>+<kbd>Z</kbd> should be kept free such that `Ż` can be produced. So issue #50053 is indeed a duplicate of this issue.

**Comment by Tekbr** [comment]
-- Sorry for the English, I used Google Translator --

I know the team has put a lot of resources into the VSCode. So much that it has much prominence in Windows, OS X and Linux and in several languages and not only in English.

And because of this dimension, I think you can already pay attention to Keyboard Layout for a variety of languages. Regardless of the number of votes. As I see it is a trivial resource. It already has a lot of other cases related precisely because of this problem.

It is very bad you look for a certain type of shortcut (which is in US standard) and when using in your layout (in my case PT-BR2) it does not match or be seen otherwise.

This affects both the VSCode and the extensions used in it.

But I want to congratulate the team for their great work.

Thanks for the attention.

// ping @alexandrudima

**Comment by jrpub** [comment]
Refering to #32473 (which has been closed), to open terminal on many french laptops: 

ctrl + ù 

Sorry if it has been already disclosed, I didn't read everything, just wanted to put the tips.

**Comment by kingma-sbw** [comment]
It is possible to find the key bindings under 
File, Preferences, Keyboard shortcuts.

What is annoying is that Help, Keyboard Shortcuts Reference points to a pdf that has no relation to the actual key bindings. I

**Comment by fleps** [comment]
I have a small programable keyboard that uses BootMapper Client to setup keybinds. It's configured to send the normal extra "F" keys tradicional keyboards don't have like F20, 21, 22, 23, 24, which are all valid keys.
For some reason VSCode does not detect these keys (but any other software does)

If I try to use the F20 key, what I get is an "unknown"
Using the debugger, I get this:

/ Received  keydown event - modifiers: [], code: F20, keyCode: 131, key: F20
| Converted keydown event - modifiers: [], code: F20, keyCode: 0 ('unknown')
\ Keyboard event cannot be dispatched in keydown phase.

If I use other keys like F12, this doesn't happen, even if I program the F12 to come from the macro keyboard.

Why VSCode just don't use the KeyCode? Why does it need to "convert" it??

It seems we had a setting in the past that fixed this on Windows ( "keyboard.dispatch": "keyCode" ) but now this setting is not valid anymore.

Again, why? Why overcomplicate such a simple thing for valid keyCodes that exist since forever?

**Comment by Lemmingh** [comment]
@fleps
Not sure what do you mean. But the following might give some clues.

> Why VS Code just don't use the `keyCode`? Why does it need to "convert" it?

From an article written in 2017 <https://github.com/microsoft/vscode/wiki/Keybinding-Issues>

> any piece of code that ends up depending on `e.keyCode` will ultimately be incorrect on various keyboard layouts on macOS/Linux.
>
> On Windows, VS Code dispatches on `e.keyCode`.
>
> On macOS and Linux, VS Code dispatches on `e.code`.

---

> `keyboard.dispatch`

It is for macOS and Linux, never Windows: 822ab91b892c0d6caa398dc28a9cb62f7f913725

<https://github.com/microsoft/vscode/blob/436725c584e4422e6764a3c19970f7d1d7f6971c/src/vs/workbench/services/keybinding/browser/keybindingService.ts#L845-L852>

**Comment by Lemmingh** [comment]
I guess I get you idea now. The "F20" case seems to be an issue from Chrome.

<https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent/code/code_values>

<https://github.com/microsoft/vscode/blob/71222493f3fa6e9489923b4437c6723c5a65b908/src/vs/workbench/services/keybinding/common/windowsKeyboardMapper.ts#L619>

**Comment by Fred-Vatin** [comment]
This could be solved if VSCode had an option to interpret characters as valid shortkeys, just like editors like **VIM**, **Helix**, **Zed** and so on.

For instance, VSCode (on Windows at least) interprets `AltGr` as `Ctrl+Alt`. So any layout using `AltGr+[key]` to produce a special character could conflict with the shortkey `Ctrl+Alt+[key]` instead of outputting the desired char.

The workaround is to create a layout that replaces `AltGr` with `Kana`. I linked `Kana` to `OEM_8` with **KbdEdit**. Now `Kana+[key]` will output the desired character without conflict. But it is impossible to use this character as a shortkey.

For instance, in my layout, I have `Kana+F` to output `/`. Here is how this is interpreted by VSCode:

```
2026-07-29 05:24:42.237 [info] [Window] [KeybindingService]: / Soft dispatching keyboard event
2026-07-29 05:24:42.237 [info] [Window] [KeybindingService]: | Resolving OEM_8
2026-07-29 05:24:42.237 [info] [Window] [KeybindingService]: \ No keybinding entries.
2026-07-29 05:24:42.238 [info] [Window] [KeybindingService]: / Received  keydown event - modifiers: [], code: AltRight, keyCode: 223, key: Unidentified
2026-07-29 05:24:42.238 [info] [Window] [KeybindingService]: | Converted keydown event - modifiers: [], code: AltRight, keyCode: 96 ('OEM_8')
2026-07-29 05:24:42.238 [info] [Window] [KeybindingService]: | Resolving OEM_8
2026-07-29 05:24:42.242 [info] [Window] [KeybindingService]: \ No keybinding entries.
2026-07-29 05:24:42.253 [info] [Window] [KeybindingService]: / Soft dispatching keyboard event
2026-07-29 05:24:42.253 [info] [Window] [KeybindingService]: | Resolving F
2026-07-29 05:24:42.253 [info] [Window] [KeybindingService]: \ From 13 keybinding entries, no when clauses matched the context.
2026-07-29 05:24:42.253 [info] [Window] [KeybindingService]: / Received  keydown event - modifiers: [], code: KeyF, keyCode: 70, key: /
2026-07-29 05:24:42.253 [info] [Window] [KeybindingService]: | Converted keydown event - modifiers: [], code: KeyF, keyCode: 36 ('F')
2026-07-29 05:24:42.253 [info] [Window] [KeybindingService]: | Resolving F
2026-07-29 05:24:42.254 [info] [Window] [KeybindingService]: \ From 13 keybinding entries, no when clauses matched the context.
```

**Please, add an option to detect `key` as keybinding when it is identified.**

This would make life so much easier for so many users who rely on custom or specific keyboard layouts, or Kanata, which sends characters via its `Unicode` function that are also detected by VSCode as a `key`.

It is a pain to have to use other keybindings on VSCode because of this when switching with other editors constantly.
